src/DynamoCore/Models/DynamoModelCommands.cs

```diff
@@ -265,6 +265,7 @@ private void SelectModelImpl(SelectModelCommand command)
                     if (!command.Modifiers.HasFlag(ModifierKeys.Shift) && command.ModelGuids.Count() == 1)
                         ClearSelectionAndRecordUndo();
 
+                    WorkspaceModel.RecordModelsForModification(new List<ModelBase>() { model }, CurrentWorkspace.UndoRecorder);
```

Comment:
It seems in the future everytime we add model to `Selection`, we will need to record it for Undo? If Yes, maybe it is worth to wrap this and the following line into a utility function, similar to the `ClearSelectionAndRecordUndo`.

Reply:
Yes, we need this. Ok, I will set this up. Thanks!